src/Microsoft.Data/PrimitiveColumn.cs

```diff
@@ -106,13 +122,23 @@ public override BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapInd
             {
                 if (mapIndices.DataType != typeof(long))
                     throw new ArgumentException(Strings.MismatchedValueType + " PrimitiveColumn<long>", nameof(mapIndices));
-                if (mapIndices.Length != Length)
+                if (mapIndices.Length >= Length)
                     throw new ArgumentException(Strings.MismatchedColumnLengths, nameof(mapIndices));
```

Comment:
Does `MismatchedColumnLengths` provide a good error message here now? Before it was just different lengths, but now the check is only for greater than or equals.

Same applies to the below usages of MismatchedColumnLengths.